notebooks/Interpretability - Text Explainers.ipynb

```diff
@@ -0,0 +1 @@
1866fb26"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"data":"","errorSummary":"","metadata":{},"errorTraceType":null,"type":"ipynbError","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>"]}}],"execution_count":0},{"cell_type":"markdown","source":["Then we use the Kernel SHAP text explainer to explain the model's predicted probability for a given observation.\n\n> Notice that we drop the base value from the SHAP output before displaying the SHAP values. The base value is the model output for an empty string."],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"de0ff3a3-fc93-4769-8b98-2fab81467fcc"}}},{"cell_type":"code","source":["shap = TextSHAP()\\\n        .setModel(model)\\\n        .setOutputCol(\"shaps\")\\\n        .setInputCol(\"text\")\\\n        .setTargetCol(\"probability\")\\\n        .setTargetClasses([1])\\\n        .setTokensCol(\"tokens\")\\\n        .setNumSamples(5000)\n\nshap_results = shap.transform(explain_instances)\\\n            .select(\"tokens\", \"shaps\", \"r2\", \"probability\", \"text\")\\\n            .withColumn(\"probability\", vec_access(\"probability\", lit(1)))\\\n            .withColumn(\"shaps\", vec2array(col(\"shaps\").getItem(0)))\\\n            .withColumn(\"shaps\", slice(col(\"shaps\"), lit(2), size(col(\"shaps\"))))\\\n            .withColumn(\"r2\", vec_access(\"r2\", lit(0)))\\\n            .withColumn(\"tokens_shaps\", arrays_zip(\"tokens\", \"shaps\"))\n\ndisplay(shap_results.select(\"probability\", \"r2\", \"tokens_shaps\", \"text\").orderBy(col(\"probability\").desc()))"],"metadata":{"application/vnd.databricks.v1+cell":{"title":"","showTitle":false,"inputWidgets":{},"nuid":"9d3fd01d-f140-465e-ae53-d3b25f246e4d"}},"outputs":[{"output_type":"display_data","metadata":{"application/vnd.databricks.v1+output":{"data":"","errorSummary":"","metadata":{},"errorTraceType":null,"type":"ipynbError","arguments":{}}},"output_type":"display_data","data":{"text/html":["<style scoped>\n  .ansiout {\n    display: block;\n    unicode-bidi: embed;\n    white-space: pre-wrap;\n    word-wrap: break-word;\n    word-break: break-all;\n    font-family: \"Source Code Pro\", \"Menlo\", monospace;;\n    font-size: 13px;\n    color: #555;\n    margin-left: 4px;\n    line-height: 19px;\n  }\n</style>"]}}],"execution_count":0}],"metadata":{"application/vnd.databricks.v1+notebook":{"notebookName":"Interpretability - Text Explainers","dashboards":[],"notebookMetadata":{"pythonIndentUnit":2},"language":"python","widgets":{},"notebookOrigID":913802417841163}},"nbformat":4,"nbformat_minor":0}
```

Comment:
nit: might want to use kwarg syntax for text explainer as the line is a bit long

Reply:
Done.